src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -193,6 +201,10 @@ private SnowflakeTableRow processRecord(SinkRecord record) {
       meta.put(SCHEMA_ID, valueContent.getSchemaID());
     }
 
+    if (metadataConfig.connectorTimeFlag) {
+      meta.put(CONNECTOR_TIME, clock.instant().toEpochMilli());
```

Comment:
could we get the timestamp of when insertRow(s) to the streaming SDK happened?

Reply:
This method is called right before pushing data to ingest-sdk. The metadata is written as json immediately, so if we want to call it in the exact place where there rows are being pushed further, refactoring of the record service is needed.

That's the call stack:
`TopicPartitionChannel.insertRowsWithFallback` -> `InsertRowsApiResponseSupplier` -> `StreamingBuffer.getData()` -> `RecordService.getProcessedRecordForStreamingIngest` -> `RecordService.processRecord`.